src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -286,6 +312,116 @@ export class HaConfigDeviceDashboard extends LitElement {
+                          "number",
+                          this._numHiddenDevices
+                        ) +
+                        ")"
+                      : ""}
+                  </paper-tooltip>
+                </div>`
+              : `${this.hass.localize(
+                  "ui.panel.config.filtering.filtering_by"
+                )} ${activeFilters.join(", ")}
+                    ${
+                      this._numHiddenDevices
+                        ? "(" +
+                          this.hass.localize(
+                            "ui.panel.config.devices.picker.filter.hidden_devices",
+                            "number",
+                            this._numHiddenDevices
+                          ) +
+                          ")"
+                        : ""
+                    }
+                    `}
+            <mwc-button @click=${this._clearFilter}
+              >${this.hass.localize(
+                "ui.panel.config.filtering.clear"
+              )}</mwc-button
+            >
+          </div>`
+        : ""}
+      ${this._numHiddenDevices && !activeFilters
+        ? html`<div class="active-filters">
+            ${this.narrow
+              ? html` <div>
+                  <ha-icon icon="hass:filter-variant"></ha-icon>
+                  <paper-tooltip animation-delay="0" position="left">
+                    ${this.hass.localize(
+                      "ui.panel.config.devices.picker.filter.hidden_devices",
+                      "number",
+                      this._numHiddenDevices
+                    )}
+                  </paper-tooltip>
+                </div>`
+              : `${this.hass.localize(
+                  "ui.panel.config.devices.picker.filter.hidden_devices",
+                  "number",
+                  this._numHiddenDevices
+                )}`}
+            <mwc-button @click=${this._showAll}
+              >${this.hass.localize(
+                "ui.panel.config.devices.picker.filter.show_all"
+              )}</mwc-button
+            >
+          </div>`
+        : ""}
+      <ha-button-menu corner="BOTTOM_START" multi>
+        <mwc-icon-button
+          slot="trigger"
+          .label=${this.hass!.localize(
+            "ui.panel.config.devices.picker.filter.filter"
+          )}
+          .title=${this.hass!.localize(
+            "ui.panel.config.devices.picker.filter.filter"
+          )}
+        >
+          <ha-svg-icon .path=${mdiFilterVariant}></ha-svg-icon>
+        </mwc-icon-button>
+        <mwc-list-item
+          @request-selected="${this._showDisabledChanged}"
+          graphic="control"
+          .selected=${this._showDisabled}
+        >
+          <ha-checkbox
+            slot="graphic"
+            .checked=${this._showDisabled}
+          ></ha-checkbox>
+          ${this.hass!.localize(
+            "ui.panel.config.devices.picker.filter.show_disabled"
+          )}
+        </mwc-list-item>
+      </ha-button-menu>
+    `;
+
```

Comment:
This is copied from `/src/panels/config/entities/ha-config-entities.ts`, could be moved to a separate component to avoid duplicating the code.